c_emulator/riscv_sim.c

```diff
@@ -378,7 +378,12 @@ static int process_args(int argc, char **argv)
       set_config_print(optarg, true);
       break;
     case 'l':
-      insn_limit = atoi(optarg);
+      char *temp;
+      insn_limit = (uint64_t)strtoull(optarg, &temp, 0);
+      if (*temp != '\0') {
+        fprintf(stderr, "invalid instruction limit %s\n", optarg);
+        exit(1);
+      }
```

Comment:
To be robust you want something like:
```c
    case 'l': {
      char *p;
      unsigned long long val;
      errno = 0;
      val = strtoull(optarg, &p, 0);
      if (*p != '\0' || val > UINT64_MAX || (val == ULLONG_MAX && errno == ERANGE)) {
        fprintf(stderr, "invalid instruction limit %s\n", optarg);
        exit(1);
      }
      break;
    }
```
Note the curly braces too for scoping, case statements are just glorified labels.

Reply:
thanks, done